### PR TITLE
Contributing: Updates contribution guidance. Adds some related agentic commands.

### PR DESCRIPTION
**Pull request by humancompanion-usds**
## Summary

* Adds some agentic AI commands that I'm in the process of refining.
* Uses one of those prompts to update the contribution guidance in prep for deprecating the Wiki for contribution guidelines.

## Issues

* Amends #[#3561]

[Open Preview Environment](https://dev-design.va.gov/4413)

### CODE REVIEW

File: .github/chatmodes/4.1.chatmode.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+Take your time and think through every step - remember to check your solution rigorously and watch out for boundary cases, especially with the changes you made. Your solution must be perfect. If not, continue working on it. At the end, you must test your code rigorously using the tools provided, and do it many times, to catch all edge cases. If it is not robust, iterate more and make it perfect. Failing to test your code sufficiently rigorously is the NUMBER ONE failure mode on these types of tasks; make sure you handle all edge cases, and run existing tests if they are provided.
+
+You MUST plan extensively before each function call, and reflect extensively on the outcomes of the previous function calls. DO NOT do this entire process by making function calls only, as this can impair your ability to solve the problem and think insightfully.
+
+# Workflow
+
+1. Understand the problem deeply. Carefully read the issue and think critically about what is required.
```
Comment (jamigibbs) [inline .github/chatmodes/4.1.chatmode.md:25]: It feels like the intention is _not_ to have any human interaction and to allow it to be fully autonomous but I've found that by adding the following prompt in my own interactions, it has helped us get to a correct or better solution faster. It does require some up front human assistance though to answer the clarifying questions.

```
_Before you answer or solve the problem, assess the uncertainty 
of your response. If it's greater than 0.1, ask me clarifying questions 
until the uncertainty is 0.1 or lower._
```

**Review by babsdenney — APPROVED** [review]
